src/private/MDMTransaction+Private.h

```diff
@@ -19,8 +19,18 @@
 @protocol MDMPlan;
 
 @interface MDMTransactionLog : NSObject
-@property(nonatomic) NSArray<id<MDMPlan>> *plans;
-@property(nonatomic) id target;
+
+@property(nonatomic, copy) NSArray<id<MDMPlan>> *plans;
+@property(nonatomic, strong) id target;
+@property(nonatomic, copy) NSString *name;
+@property(nonatomic, assign, readonly, getter=isRemoval) BOOL removal;
+
+- (instancetype)initWithPlans:(NSArray<NSObject<MDMPlan> *> *)plan target:(id)target name:(NSString *)name removal:(BOOL)removal;
+
+- (instancetype)initWithPlans:(NSArray<NSObject<MDMPlan> *> *)plans target:(id)target name:(NSString *)name;
+
+- (instancetype)initWithTarget:(id)target name:(NSString *)name;
```

Comment:
- [ ] If possible, revert these changes so as not to add complexity to this commit.